typing/printtyp.ml

```diff
@@ -1435,6 +1435,26 @@ let explanation unif t3 t4 ppf =
       end
   | _ -> ()
 
+
+let warn_on_missing_def env ppf t =
+  let is_predef = function
+    | Pident ident' -> List.exists (fun (_,ident) -> ident = ident') Predef.builtin_idents
+    | _ -> false in
+  let missing_def = function
+    | Tconstr (p,_,_) when is_predef p -> None
```

Comment:
Why do you need a special case for Predef? They are already included in env, so the following case should be sufficient.